src/librustc_traits/lowering.rs

```diff
@@ -101,14 +101,15 @@ impl<'tcx> Lower<PolyDomainGoal<'tcx>> for ty::Predicate<'tcx> {
     }
 }
 
-/// Transforms an existing goal into a FromEnv goal.
-///
 /// Used for lowered where clauses (see rustc guide).
-trait IntoFromEnvGoal {
+trait IntoGoal {
+    // Transforms an existing goal into a FromEnv goal.
     fn into_from_env_goal(self) -> Self;
+    // Transforms an existing goal into a WellFormed goal.
+    fn into_wellformed_goal(self) -> Self;
```

Comment:
Very nit: `well_formed` is how we snake-casify `WellFormed` in chalk, so I'd like `into_well_formed_goal` for consistency :)